lib/internal/test_runner/coverage.js

```diff
@@ -464,23 +483,24 @@ class TestCoverage {
       coverageExcludeGlobs: excludeGlobs,
       coverageIncludeGlobs: includeGlobs,
     } = this.options;
-    // This check filters out files that match the exclude globs.
-    if (excludeGlobs?.length > 0) {
-      for (let i = 0; i < excludeGlobs.length; ++i) {
-        if (matchesGlob(relativePath, excludeGlobs[i]) ||
-            matchesGlob(absolutePath, excludeGlobs[i])) return true;
-      }
-    }
 
     // This check filters out files that do not match the include globs.
     if (includeGlobs?.length > 0) {
       for (let i = 0; i < includeGlobs.length; ++i) {
-        if (matchesGlob(relativePath, includeGlobs[i]) ||
-            matchesGlob(absolutePath, includeGlobs[i])) return false;
+        if (glob(relativePath, includeGlobs[i]) ||
+            glob(absolutePath, includeGlobs[i])) return false;
       }
       return true;
     }
 
+    // This check filters out files that match the exclude globs.
+    if (excludeGlobs?.length > 0) {
+      for (let i = 0; i < excludeGlobs.length; ++i) {
+        if (glob(relativePath, excludeGlobs[i]) ||
+            glob(absolutePath, excludeGlobs[i])) return true;
```

Comment:
Nit: this is difficult to grok

```suggestion
        if (
          glob(relativePath, excludeGlobs[i]) ||
          glob(absolutePath, excludeGlobs[i])
        ) {
          return true;
        }
```

Reply:
I'll address this in the next commit 🚀